classes/conversion.php

```diff
@@ -159,25 +159,34 @@ private function create_conversion(\stored_file $file) : void {
         $now = time();
         $convid = 0;
 
-        $convrecord = new \stdClass();
-        $convrecord->pathnamehash = $file->get_pathnamehash();
-        $convrecord->contenthash = $file->get_contenthash();
-        $convrecord->status = $this::CONVERSION_ACCEPTED;
-        $convrecord->transcribe = $this->config->transcribe;
-        $convrecord->rekog_label = $this->config->detectlabels;
-        $convrecord->rekog_moderation = $this->config->detectmoderation;
-        $convrecord->rekog_face = $this->config->detectfaces;
-        $convrecord->rekog_person = $this->config->detectpeople;
-        $convrecord->detect_sentiment = $this->config->detectsentiment;
-        $convrecord->detect_phrases = $this->config->detectphrases;
-        $convrecord->detect_entities = $this->config->detectentities;
-        $convrecord->timecreated = $now;
-        $convrecord->timemodified = $now;
+        $cnvrec = new \stdClass();
+        $cnvrec->pathnamehash = $file->get_pathnamehash();
+        $cnvrec->contenthash = $file->get_contenthash();
+        $cnvrec->status = $this::CONVERSION_ACCEPTED;
+        $cnvrec->transcoder_status = $this::CONVERSION_ACCEPTED;
+        $cnvrec->transcribe_status =
```

Comment:
Could this be condensed to something like a mapping array: 
```php
$settingidconversionfields = ['transcribe_status' => 'transcribe', 'rekog_label_status' => 'detectlabels'...]
``` 
and a loop of:
```php
for each ($settingidconversionfields as $settingid => $conversionfield) {
    $cnvrec->$conversionfield = 
        $this->config->$settingid == 1 ? $this::CONVERSION_ACCEPTED : $this::CONVERSION_NOT_FOUND
}
```
Or is that overkill?

Reply:
good idea